response_handler_lib/http_interceptor.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+import requests
+from requests.exceptions import HTTPError
+
+from response_handler_lib.error_codes import PredefinedErrorCodes
+from response_handler_lib.response import Response
+
+
+class HTTPInterceptor:
+    def __init__(self, include_where=False):
+        self.include_where = include_where
+        self.response = Response()
+
+    def request(self, method, url, **kwargs):
+        resp = None
+        try:
+            resp = requests.request(method, url, **kwargs)
+            resp.raise_for_status()
+            if resp.status_code != 204:
+                self.response.data = resp.json()
+        except HTTPError as http_err:
+            self.handle_http_error(http_err, resp)
+        except Exception as err:
+            self.handle_generic_error(err)
+        return self.response.to_json()
+
+    def handle_http_error(self, http_err, resp):
+        if resp.status_code == 400:
+            self.response.add_error(PredefinedErrorCodes.BAD_REQUEST.value)
+        elif resp.status_code == 404:
+            self.response.add_error(PredefinedErrorCodes.NOT_FOUND.value)
+        elif resp.status_code == 500:
+            self.response.add_error(PredefinedErrorCodes.INTERNAL_ERROR.value)
+        else:
+            self.response.add_error(PredefinedErrorCodes.INTERNAL_ERROR.value)
+        logging.error(f"HTTP error occurred: {http_err}")
```

Comment:
Better with match and case

**suggestion** 💭 


```suggestion
        match resp.status_code:
            case 400:
                self.response.add_error(PredefinedErrorCodes.BAD_REQUEST.value)        
            case 404:
                self.response.add_error(PredefinedErrorCodes.BAD_REQUEST.value)
            case 500:
                self.response.add_error(PredefinedErrorCodes.INTERNAL_ERROR.value)
            case _:
                self.response.add_error(PredefinedErrorCodes.INTERNAL_ERROR.value)
        logging.error(f"HTTP error occurred: {http_err}")
```

<sub><sup>I'm using [conventional comments](https://conventionalcomments.org/) formatting to improve comments readability and reader's expectations in this PR.</sup></sub>

Reply:
`match` and `case` constructs are not available in Python 3.9 as they were introduced in Python 3.10.